### PR TITLE
fix: resolve uncles hash calculation issue

### PR DESCRIPTION
**Pull request by quake**
Uncles hash is the blake2b on concatenated uncle block hashes.

### CODE REVIEW

File: util/types/src/extension/calc_hash.rs [hunk 1]
```diff
@@ -214,6 +220,78 @@ mod tests {
...
+            .raw(uncle2_raw_header)
+            .nonce(0x2f39_2d41_cd70_024fu128.pack())
+            .build();
+        let uncle2 = packed::UncleBlock::new_builder()
+            .header(uncle2_header)
+            .build();
+
+        let uncles = vec![uncle1, uncle2].pack();
+        let expect = h256!("0x0135d01f169a870bd9c92b2b37aecfa0fbfb7c1862cc176e03bb525fab0649d9");
+        assert_eq!(uncles.calc_uncles_hash(), expect.pack());
```
Comment (u2) [inline util/types/src/extension/calc_hash.rs]: I don't like this test case, there are too many hard codes, it's not maintainable.

Comment (quake) [reply util/types/src/extension/calc_hash.rs]: It's just a test vector for reference, which can be used to verify the correctness of different language implementations.

Comment (u2) [reply util/types/src/extension/calc_hash.rs]: ok

**Pull request merged**